app/controllers/subscription.js

```diff
@@ -214,6 +215,39 @@ export default Ember.Controller.extend({
     } else if (existingItem) { //delete
       this.store.unloadRecord(existingItem);
     }
+
+    // mark message read here
+    if (type === "message") {
+      var router = this.get("target");
+      var currentUrl = window.location.href.split("#").get("lastObject");
```

Comment:
What type of url we are getting here ? I think it would be better if we can fetch this in route and make it available in controller. Thoughts ?
CC: @namrataukirde

Reply:
This will return `http://localhost:4202/orders/21191/conversation`, we need this to verify that user is on messages page. and then mark the latest message `read`.

Reply:
This scenario is, when the user is chatting with stock user or vice versa. New message which has come up, we need to mark that message `read` . 
So i have written code in a way that it verifies the currentHref, with the generated messageUrl.
and if they match, i hit the mark_read method from the messages_controller.

Reply:
> I think it would be better if we can fetch this in route

That wont work for new incoming messages if user is still chatting.